### DISCUSSION TITLE
openmp in opensycl

### DISCUSSION THREAD

**Discussion by garadeaniket** [AdaptiveCpp/AdaptiveCpp#1024] [Q&A] [answered]

**Comment by illuhad** [comment]
1. It is not. Kernel code is always identified at compile time. In the generic SSCP compilation flow (which does not yet support CPU), kernel code is then lowered to backend-specific formats at runtime. Are you referring to this?
2. No it is not. In the CPU backend, SYCL constructs are mapped to OpenMP constructs to some extent. It is also possible to have both SYCL and OpenMP in the same application, but then OpenMP is not translated to SYCL, it is just natively compiled as OpenMP.

**Reply by illuhad** [reply]
It does not need to. It's the job of the OpenMP frontend to recognize OpenMP code, not Open SYCL's job.

**Reply by illuhad** [reply]
Yes. Open SYCL only cares about SYCL constructs. Other languages that clang also supports are handled by their respective language frontends and runtimes.
If you have an OpenMP code and compile it with `syclcc`, the exact same thing will happen as when compiling that code directly with the underlying OpenMP compiler such as clang.

**Reply by illuhad** [reply, marked as answer]
In clang you'd register a pragma handler for this: https://clang.llvm.org/doxygen/classclang_1_1PragmaNamespace.html
I don't know how it would work with other compilers.

**Reply by garadeaniket** [reply]
ok thankyou
